source/common/http/conn_manager_config.h

```diff
@@ -336,6 +337,13 @@ class ConnectionManagerConfig {
    */
   virtual Config::ConfigProvider* scopedRouteConfigProvider() PURE;
 
+  /**
+   * @return Router::ScopeKeyBuilder* the scope key builder to calculate the scope key.
+   * Pointer ownership is _not_ transferred to the caller of this function. This will return
+   * nullptr when scoped routing is not enabled.
+   */
+  virtual const Router::ScopeKeyBuilder* scopeKeyBuilder() PURE;
```

Comment:
From the code it looks like this pointer can never be nullptr. Is it possible to return the reference to make it explicit?

Reply:
I think not. It will not be nullptr only when scoped routing is used, just like `scopedRouteConfigProvider`.

Reply:
In this case returning OptRef may be better, as it forces the caller to validate that the returned value isn't empty/null.